eng/pipelines/templates/steps/checkout-source-dependencies.yml

```diff
@@ -0,0 +1,32 @@
+parameters:
+  - name: ProjectList
+    type: string
+  - name: CheckoutRecordings
+    type: boolean
+    default: false
+
+steps:
+  - task: PythonScript@0
+    displayName: 'Generate directories variable for sparse checkout'
+    inputs:
+      scriptPath: 'eng/scripts/generate_from_source_pom.py'
+      arguments: '--set-pipeline-variable --project-list ${{ parameters.ProjectList }}'
+      workingDirectory: '$(System.DefaultWorkingDirectory)'
+
+  - pwsh: |
+      $paths = ('$(CheckoutDirectories)' | ConvertFrom-Json) -join ' '
+      $javaPaths = $paths | ForEach-Object { $_ + '/**/*.java' }
+      Invoke-Expression -Command "git sparse-checkout set eng '**/*.xml' '**/*.md' '**/test-resources.json' $javaPaths"
```

Comment:
I have to say this template seems a little scary to me and trying to understand it will likely be trouble in the future.

For example why do we need to add eng, "**/*.xml", "**/*.md" again here? Isn't that already included in the sparse checkout template?

Reply:
I'm still testing this and have switched it to something a little simpler:

```
  - pwsh: |
      $paths = ('$(CheckoutDirectories)' | ConvertFrom-Json) -join ' '
      $command = "git sparse-checkout set eng '**/*.xml' '**/*.md' '**/test-resources.json' $paths"
      if (!$${{ parameters.CheckoutRecordings }}) {
        $command += " '!/**/session-records/*/.json'"
        $command += " '!/**/test-recordings/*/.json'"
      }
      Write-Host $command
      Invoke-Expression -Command $command
      Write-Host "Set sparse checkout paths to:"
      Get-Content .git/info/sparse-checkout
    displayName: Sparse checkout module paths
```

Reply:
I think I can move the xml/md out of here.

Reply:
Perhaps I should also do away with `CheckoutRecordings` and just pass in the exclusion paths from the `live.tests.yml` file.

Reply:
Another possibility is to remove the variable passing, and instead have the pom scanner script dump out a formatted sparse checkout file that we can just drop into place? That would simplify the yaml quite a bit.